<?xml version='1.0' encoding='UTF-8'?>
<robot name="stretch_description">
  <link name="link_wrist_yaw_bottom">
    <inertial>
      <origin xyz="-0.012839101377342 -0.0382787718640742 -0.0228400332263617" rpy="0.0 0.0 0.0"/>
      <mass value="0.0988906816399982"/>
      <inertia ixx="2.60067866573596e-05" ixy="-6.73176267521354e-06" ixz="-2.43476436723672e-06" iyy="5.99482946819923e-06" iyz="-3.39642410492401e-06" izz="2.56907114334732e-05"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_wrist_yaw_bottom.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_wrist_yaw_bottom.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_wrist_pitch">
    <inertial>
      <origin xyz="-0.00310609611067142 -0.0150777141465843 0.0204734587925901" rpy="0.0 0.0 0.0"/>
      <mass value="0.0701267146295583"/>
      <inertia ixx="2.55965614980905e-06" ixy="-1.47551515167608e-06" ixz="-6.31436085977252e-08" iyy="3.43968637386282e-06" iyz="-4.17813567208843e-07" izz="4.53568668211393e-06"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_wrist_pitch.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_wrist_pitch.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_wrist_roll">
    <inertial>
      <origin xyz="9.63118473862323e-15 -6.38378239159465e-15 0.00768048802649798" rpy="0.0 0.0 0.0"/>
      <mass value="0.00585666394358811"/>
      <inertia ixx="2.55965614980905e-06" ixy="-1.47551515167608e-06" ixz="-6.31436085977252e-08" iyy="3.43968637386282e-06" iyz="-4.17813567208843e-07" izz="4.53568668211393e-06"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_wrist_roll.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_wrist_roll.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_straight_gripper">
    <inertial>
      <origin xyz="0.00150764845432383 -0.00711581846201287 0.0399737901417758" rpy="0.0 0.0 0.0"/>
      <mass value="0.0496384234458284"/>
      <inertia ixx="5.61461154156397e-06" ixy="8.29518962984231e-07" ixz="-2.41382921888194e-06" iyy="1.11504692003467e-05" iyz="9.76174898123369e-07" izz="6.63803357903882e-06"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_straight_gripper.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_straight_gripper.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_grasp_center"/>
  <link name="base_link">
    <inertial>
      <origin xyz="-0.109461304328163 -0.000741018909047708 0.0914915269429946" rpy="0.0 0.0 0.0"/>
      <mass value="1.0723782659782"/>
      <inertia ixx="0.00310580907710135" ixy="1.5182848191076e-06" ixz="0.00041690466732394" iyy="0.00433798719991832" iyz="1.33487716258445e-05" izz="0.0037204727467362"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/base_link.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/base_link.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_mast">
    <inertial>
      <origin xyz="0.00755818572975822 0.773971284176834 0.00647313086620024" rpy="0.0 0.0 0.0"/>
      <mass value="0.749143203376401"/>
      <inertia ixx="0.0709854511954588" ixy="-0.00433428742758457" ixz="-0.000186110788697573" iyy="0.000437922053342648" iyz="-0.00288788257713431" izz="0.071104808501661"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_mast.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_mast.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_lift">
    <inertial>
      <origin xyz="-0.0547636346252869 0.0380939689802074 0.00933159404448308" rpy="0.0 0.0 0.0"/>
      <mass value="0.444820831236007"/>
      <inertia ixx="0.000567192946271367" ixy="-0.00024607534789508" ixz="-0.000128108442260574" iyy="0.00127108364478659" iyz="7.61916855070339e-06" izz="0.00137296635099006"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_lift.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_lift.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_arm_l4">
    <inertial>
      <origin xyz="-1.61247276464493e-06 1.54715589201215e-06 -0.0953252706040035" rpy="0.0 0.0 0.0"/>
      <mass value="0.0676525455789735"/>
      <inertia ixx="0.000122002023066796" ixy="7.29593356230121e-13" ixz="9.63170353371061e-09" iyy="0.000122002023127152" iyz="-9.24154949889042e-09" izz="1.46043439809638e-12"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l4.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l4.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_arm_l3">
    <inertial>
      <origin xyz="-3.34873468621488e-07 4.7411336306924e-07 -0.0974069598289041" rpy="0.0 0.0 0.0"/>
      <mass value="0.0626921047058405"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l3.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l3.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_arm_l2">
    <inertial>
      <origin xyz="-3.31043560508615e-07 3.13422303999111e-07 -0.097255883541891" rpy="0.0 0.0 0.0"/>
      <mass value="0.0569074368576238"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l2.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l2.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_arm_l1">
    <inertial>
      <origin xyz="-3.36876635420591e-07 3.05876091066803e-07 -0.0969117032232756" rpy="0.0 0.0 0.0"/>
      <mass value="0.0511763619538321"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l1.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l1.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_arm_l0">
    <inertial>
      <origin xyz="0.0230715049668191 -0.00195115833662818 -0.0360646347160285" rpy="0.0 0.0 0.0"/>
      <mass value="0.101241250325294"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l0.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l0.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <link name="link_wrist_yaw">
    <inertial>
      <origin xyz="-8.02795716750571e-06 1.93811447768422e-05 0.018946700595851" rpy="0.0 0.0 0.0"/>
      <mass value="0.0405398981326229"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_wrist_yaw.STL"/>
      </geometry>
      <material name=""/>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_wrist_yaw.STL"/>
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
    </collision>
  </link>
  <joint name="joint_wrist_yaw_bottom" type="fixed">
    <parent link="link_wrist_yaw"/>
    <child link="link_wrist_yaw_bottom"/>
    <axis xyz="nan nan nan"/>
    <origin xyz="0.0 0.0 0.0" rpy="-3.14159265358979 1.13367999021379e-14 1.57079632679489"/>
  </joint>
  <joint name="joint_wrist_pitch" type="revolute">
    <limit effort="100.0" velocity="1.0" lower="-1.57" upper="0.56"/>
    <parent link="link_wrist_yaw_bottom"/>
    <child link="link_wrist_pitch"/>
    <axis xyz=" 0.  0. -1."/>
    <origin xyz="0.0 -0.0195500000000002 -0.0247499999999984" rpy="1.5707963267949 -8.12895570882604e-15 -3.14159265358979"/>
  </joint>
  <joint name="joint_wrist_roll" type="revolute">
    <limit effort="100.0" velocity="1.0" lower="-3.14" upper="3.14"/>
    <parent link="link_wrist_pitch"/>
    <child link="link_wrist_roll"/>
    <axis xyz="0. 0. 1."/>
    <origin xyz="-0.0188587444076125 -0.0239999999998942 0.01955" rpy="3.14159265358979 1.5707963267948966 0.0"/>
  </joint>
  <joint name="joint_straight_gripper" type="fixed">
    <parent link="link_wrist_roll"/>
    <child link="link_straight_gripper"/>
    <axis xyz="nan nan nan"/>
    <origin xyz="0.0 0.0 0.0155" rpy="3.54987407349455e-30 3.24021254484265e-20 -3.14159265358979"/>
  </joint>
  <joint name="joint_grasp_center" type="fixed">
    <parent link="link_straight_gripper"/>
    <child link="link_grasp_center"/>
    <axis xyz="1. 0. 0."/>
    <origin xyz="0.0 0.0 0.23" rpy="-1.5707963267949 -1.5707963267948966 0.0"/>
  </joint>
  <joint name="joint_mast" type="fixed">
    <parent link="base_link"/>
    <child link="link_mast"/>
    <axis xyz="nan nan nan"/>
    <origin xyz="-0.07237645954942834 0.13471307674741795 0.03137898879007643" rpy="1.569197910302871 0.016019836014518063 -0.0037670252534494344"/>
  </joint>
  <joint name="joint_lift" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="1.1"/>
    <parent link="link_mast"/>
    <child link="link_lift"/>
    <axis xyz="0. 0. 1."/>
    <origin xyz="-0.0369217062323472 0.165471199999996 -0.000341653286793524" rpy="-1.57079632679552 1.5615431375291973 -6.2942004366467e-13"/>
  </joint>
  <joint name="joint_arm_l4" type="fixed">
    <parent link="link_lift"/>
    <child link="link_arm_l4"/>
    <axis xyz="nan nan nan"/>
    <origin xyz="-0.25352912668384553 -0.0022772149299145534 -0.001143373145064658" rpy="1.5717129650424848 0.039161718095026554 -1.549959798553948"/>
  </joint>
  <joint name="joint_arm_l3" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="0.13"/>
    <parent link="link_arm_l4"/>
    <child link="link_arm_l3"/>
    <axis xyz="0. 0. 1."/>
    <origin xyz="0.0 0.0 0.013" rpy="3.5742e-16 3.2123e-16 -2.4565e-16"/>
  </joint>
  <joint name="joint_arm_l2" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="0.13"/>
    <parent link="link_arm_l3"/>
    <child link="link_arm_l2"/>
    <axis xyz="0. 0. 1."/>
    <origin xyz="0.0 0.0 0.013" rpy="2.91385731014123e-16 4.44300759504684e-17 3.25691080516352e-16"/>
  </joint>
  <joint name="joint_arm_l1" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="0.13"/>
    <parent link="link_arm_l2"/>
    <child link="link_arm_l1"/>
    <axis xyz="0. 0. 1."/>
    <origin xyz="0.0 0.0 0.013" rpy="7.41596560408007e-17 1.33876788296791e-16 -2.44545873596083e-16"/>
  </joint>
  <joint name="joint_arm_l0" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="0.13"/>
    <parent link="link_arm_l1"/>
    <child link="link_arm_l0"/>
    <axis xyz="0. 0. 1."/>
    <origin xyz="0.0 0.0 -0.0137499999999938" rpy="-3.05317390622457e-16 -4.23236224076729e-16 -2.68425135229209e-17"/>
  </joint>
  <joint name="joint_wrist_yaw" type="revolute">
    <limit effort="100.0" velocity="1.0" lower="-1.75" upper="4.0"/>
    <parent link="link_arm_l0"/>
    <child link="link_wrist_yaw"/>
    <axis xyz=" 0.  0. -1."/>
    <origin xyz="0.083 -0.03075 0.0" rpy="1.5708 4.2595e-14 2.6415e-15"/>
  </joint>
  <material name="aluminum">
    <color rgba="0.5 0.5 0.5 1. "/>
  </material>
  <material name="plastic">
    <color rgba="0.1 0.1 0.1 1. "/>
  </material>
  <material name="">
    <color rgba="0.79215686 0.81960784 0.93333333 1.        "/>
  </material>
</robot>
